sdk/storage/Azure.Storage.Queues/src/QueueMessageEncoding.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Storage.Queues.Models;
+
+namespace Azure.Storage.Queues
+{
+    /// <summary>
+    /// Determines how <see cref="QueueMessage.Body"/> is represented in HTTP requests and responses.
+    /// </summary>
+    public enum QueueMessageEncoding
+    {
+        /// <summary>
+        /// The <see cref="QueueMessage.Body"/> is prepresented as UTF8 encoded string in HTTP requests and responses.
+        /// </summary>
+        UTF8 = 0,
+
+        /// <summary>
+        /// The <see cref="QueueMessage.Body"/> is prepresented as Base64 encoded string in HTTP requests and responses.
+        /// </summary>
+        Base64 = 1,
```

Comment:
Add a <remarks> section talking about interop with older versions of the library?

Reply:
Added. Turns out V11 and prior had a switch to control that behavior but it was just boolean and was opposite to what we're doing here.

Reply:
Sorry, I meant something more like remarks:

```C#
/// <summary>
/// The <see cref="QueueMessage.Body"/> is represented as Base64 encoded string in HTTP requests and responses.
/// </summary>
/// <remarks>
/// This was the default behavior in the prior v11 library.  See
/// <see href="https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.storage.queue.cloudqueue.encodemessage?view=azure-dotnet-legacy">CloudQueue.EncodeMessage</see>.
/// Using this option can make interop with an existing application easier.
/// </remarks>
```